### PR TITLE
change invalid character list, campatible with non-latin characters

### PR DESCRIPTION
**Pull request by haiwenzhu**
This commit is aim to campatible with non-latin character like Chinese.

### CODE REVIEW

File: md-to-toc.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def to_github_anchor(title):
     anchor_name = title.strip().lower().replace(' ', '-')
 
     # Strip all invalid characters
-    anchor_name = re.sub("[^A-Za-z0-9\-_]", "", anchor_name)
+    anchor_name = re.sub("[\]\[\!\"\#\$\%\&\'\(\)\*\+\,\.\/\:\;\<\=\>\?\@\\\^\_\{\|\}\~]", "", anchor_name)
```
Comment (amaiorano) [inline md-to-toc.py]: This is a useful change, but could we make this code more clear by using Python raw strings here? Could you try with:

```python
    anchor_name = re.sub(r"[][!"#$%&'()*+,./:;<=>?@\^_{|}~]", "", anchor_name)
```

I haven't tested this, but I assume it should give the same result.

**Review by amaiorano — CHANGES_REQUESTED** [review]
Let's try with the raw string. If that works, please update the PR. I'll merge one of these in any case. Thanks!

**Comment by haiwenzhu** [comment]
Thanks for your suggestion, i tried raw string, it works with following
```python
anchor_name = re.sub(r"[\[\]\"!#$%&'()*+,./:;<=>?@\^_{|}~]", "", anchor_name)
```
And i have pushed the new change.

**Comment by amaiorano** [comment]
Thanks!

**Pull request merged**